src/nvim/regexp_nfa.c

```diff
@@ -1493,8 +1493,8 @@ static int nfa_regatom(void)
 
     default:
     {
-      long n = 0;
-      int cmp = c;
+      uint64_t n = 0;
+      uint32_t cmp = c;
```

Comment:
Do I need an assertion for this? `c` is int.

Reply:
> refactoring the other regexp functions to return uint32_t will take a while. 

Just an assert is fine. Refactoring regexp module is out of scope, not worth our time.

Reply:
Actually, does `cmp` need to be fixed? It doesn't impact `n`.